operators/trello_to_s3_operator.py

```diff
@@ -0,0 +1,145 @@
+                 since=None,
+                 before=None,
+                 *args,
+                 **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.trello_conn_id = trello_conn_id
+        self.endpoint = endpoint
+        
+        self.since = since
+        self.before = before
+        self.fields = fields
+        
+        self.s3_conn_id = s3_conn_id
+        self.s3_bucket = s3_bucket
+        self.s3_key = s3_key
+        
+        self.hook = TrelloHook(http_conn_id=self.trello_conn_id)
+
+        if self.endpoint not in (
+            'boards',
+            'actions',
+            'cards',
+            'checklists',
+            'lists',
+            'members'):
+            raise Exception('Specified endpoint not currently supported.')
+
+    def get_me(self, nested_object, extra_args):
+        """
+        Get specific nested object from 'members/me' endpoint.
+        """
+        return self.hook.run('members/me/{}'.format(nested_object), extra_args=extra_args).json() 
+    
+    def get_all(self, endpoints, extra_args={}):
+        """
+        Fetch results from multiple endpoints and
+        return an array with all the results.
+        """
+        results = []
+
+        for endpoint in  endpoints: 
+            result = self.hook.run(endpoint, fields=self.fields, extra_args=extra_args).json()
+            results += result
+
+        return results
+
+    def execute(self, context):
+        boards_endpoint = 'members/me/boards'
+        extra_args = {
+            'fields': self.fields,
+            'since': self.since, 
+            'before': self.before
+        }
+
+        if self.endpoint is 'members': 
+            organizations = self.get_me('organizations', extra_args=extra_args)
+            results = []
+            
+            if organizations:
+                for org in organizations:
+                    org_members = self.hook.run(
+                        'organizations/{}/members', extra_args=extra_args).json()
+                    results += org_members
+        elif self.endpoint is 'boards':
+            results = self.hook.run(
+                boards_endpoint, extra_args=extra_args).json()
+        else:
+            boards = self.hook.run(
+                boards_endpoint,
+                    extra_args={
+                        'fields': 'id', 
+                        'since' :self.since, 
+                        'before': self.before
+                    }).json()
+            endpoints = [
+                "boards/{}/{}".format(board['id'],
+                                      self.endpoint, 
+                                      extra_args=extra_args) for board in boards]
+            results = self.get_all(endpoints, extra_args=extra_args)
+
+        with NamedTemporaryFile("w") as tmp:
+            for result in results:
+                tmp.write(json.dumps(result) + '\n')
+
+            tmp.flush()
+
+            dest_s3 = S3Hook(s3_conn_id=self.s3_conn_id)
```

Comment:
If the file is not created, this section is no longer executed without a specific message to show. 
Maybe it would be better to move this section starting with this line outside of the `with` block, and check if the content from `with` was added to the file first. 
If it wasn't written content to the file, raise a specific `Exception`, otherwise continue exection with this section.

Reply:
Since this is a temporary file it would be deleted when `__exit__` method is called, so i can't access it outside of the `with` block

Reply:
That's right, I agree with that. Thanks for clarification.

Reply:
One thing we've been adding to new operators is using the `SkipMixin` (https://github.com/apache/incubator-airflow/blob/master/airflow/models.py#L1960) to avoid writing empty files to s3 and skip any downstream dependent tasks.

example 
```
if len(results) == 0 or results is None:
            logging.info("No records pulled from Trello.")
            downstream_tasks = context['task'].get_flat_relatives(upstream=False)
            logging.info('Skipping downstream tasks...')
            logging.debug("Downstream task_ids %s", downstream_tasks)

            if downstream_tasks:
                self.skip(context['dag_run'],
                          context['ti'].execution_date,
                          downstream_tasks)
            return True

        else:
            # Write the results to a temporary file and save that file to s3.
            with NamedTemporaryFile("w") as tmp:
                for result in results:
                    filtered_result = self.filter_fields(result)
                    tmp.write(json.dumps(filtered_result) + '\n')

                tmp.flush()

                dest_s3 = S3Hook(s3_conn_id=self.s3_conn_id)
                dest_s3.load_file(
                    filename=tmp.name,
                    key=self.s3_key,
                    bucket_name=self.s3_bucket,
                    replace=True

                )
                dest_s3.connection.close()
                tmp.close()
```